core/src/libraries/LiquidityAmounts.sol

```diff
@@ -3,66 +3,13 @@ pragma solidity 0.8.17;
 
 import {Math} from "openzeppelin-contracts/contracts/utils/math/Math.sol";
 
-function mulDiv96(uint256 x, uint256 y) pure returns (uint256 result) {
-    assembly ("memory-safe") {
-        // 512-bit multiply [prod1 prod0] = x * y. Compute the product mod 2^256 and mod 2^256 - 1, then use
-        // use the Chinese Remainder Theorem to reconstruct the 512 bit result. The result is stored in two 256
-        // variables such that product = prod1 * 2^256 + prod0.
-
-        // Least significant 256 bits of the product.
-        let prod0 := mul(x, y)
-        let mm := mulmod(x, y, not(0))
-        // Most significant 256 bits of the product.
-        let prod1 := sub(mm, add(prod0, lt(mm, prod0)))
-
-        // Make sure the result is less than `2**256`.
-        if iszero(gt(0x1000000000000000000000000, prod1)) {
-            // Store the function selector of `FullMulDivFailed()`.
-            mstore(0x00, 0xae47f702)
-            // Revert with (offset, size).
-            revert(0x1c, 0x04)
-        }
-
-        // Divide [prod1 prod0] by 2^96.
-        result := or(shr(96, prod0), shl(160, prod1))
-    }
-}
+import {square, mulDiv96, mulDiv224} from "./MulDiv.sol";
 
 /// @title LiquidityAmounts
 /// @notice Provides functions for computing liquidity amounts from token amounts and prices
 /// @author Aloe Labs, Inc.
 /// @author Modified from [Uniswap](https://github.com/Uniswap/v3-periphery/blob/main/contracts/libraries/LiquidityAmounts.sol)
 library LiquidityAmounts {
-    /// @notice Computes the amount of token0 for a given amount of liquidity and a price range
-    /// @param sqrtRatioAX96 A sqrt price representing the first tick boundary
-    /// @param sqrtRatioBX96 A sqrt price representing the second tick boundary
-    /// @param liquidity The liquidity being valued
-    /// @return amount0 The amount of token0. Will fit in a uint224 if you need it to
-    function getAmount0ForLiquidity(
-        uint160 sqrtRatioAX96,
-        uint160 sqrtRatioBX96,
-        uint128 liquidity
-    ) internal pure returns (uint256 amount0) {
-        assert(sqrtRatioAX96 <= sqrtRatioBX96);
-
-        amount0 = Math.mulDiv(uint256(liquidity) << 96, sqrtRatioBX96 - sqrtRatioAX96, sqrtRatioBX96) / sqrtRatioAX96;
-    }
-
-    /// @notice Computes the amount of token1 for a given amount of liquidity and a price range
-    /// @param sqrtRatioAX96 A sqrt price representing the first tick boundary
-    /// @param sqrtRatioBX96 A sqrt price representing the second tick boundary
-    /// @param liquidity The liquidity being valued
-    /// @return amount1 The amount of token1. Will fit in a uint192 if you need it to
-    function getAmount1ForLiquidity(
-        uint160 sqrtRatioAX96,
-        uint160 sqrtRatioBX96,
-        uint128 liquidity
-    ) internal pure returns (uint256 amount1) {
-        assert(sqrtRatioAX96 <= sqrtRatioBX96);
-
-        amount1 = mulDiv96(liquidity, sqrtRatioBX96 - sqrtRatioAX96);
-    }
-
```

Comment:
These were only used in this file, so I marked them private. To conform with styleguide, moved to bottom of file. And since `getAmountsForLiquidity` already checks `sqrtRatioAX96 < sqrtRatioBX96`, I was able to remove the extra checks.